datajunction-server/datajunction_server/api/materializations.py

```diff
@@ -288,27 +288,48 @@ async def list_node_materializations(
     like urls from the materialization service, if available.
     """
     request_headers = dict(request.headers)
-    node = await Node.get_by_name(session, node_name)
+    node = await Node.get_by_name(
+        session,
+        node_name,
+        options=[
+            joinedload(Node.revisions).options(*NodeRevision.default_load_options()),
+        ]
+        if include_all
+        else [],
+    )
     materializations = []
-    for materialization in node.current.materializations:  # type: ignore
-        if not materialization.deactivated_at or show_deleted:  # pragma: no cover
-            info = query_service_client.get_materialization_info(
-                node_name,
-                node.current.version,  # type: ignore
-                node.type,  # type: ignore
-                materialization.name,  # type: ignore
-                request_headers=request_headers,
-            )
-            if materialization.strategy != MaterializationStrategy.INCREMENTAL_TIME:
-                info.urls = [info.urls[0]]
-            materialization_config_output = MaterializationConfigOutput.from_orm(
-                materialization,
-            )
-            materialization = MaterializationConfigInfoUnified(
-                **materialization_config_output.dict(),
-                **info.dict(),
-            )
-            materializations.append(materialization)
+    if include_all:  # Get all materializations for all node revisions
+        for node_revision in node.revisions:  # type: ignore
+            for materialization in node_revision.materializations:
+                materializations.append(
+                    get_materialization_info(
+                        query_service_client=query_service_client,
+                        node_revision=node_revision,
+                        materialization=materialization,
+                        request_headers=request_headers,
+                    ),
+                )
+    else:  # Get just the active materializations on the current node revision
+        for materialization in node.current.materializations:  # type: ignore
+            if not materialization.deactivated_at:  # pragma: no cover
```

Comment:
I think this code now would not give you the option of seeing just the current materialization regardless of their status. I wonder if it would be cleaner to have two params on this whole method:
- `show_deleted` as before (although I think `show_inactive` sounds better)
- `include_all_revisions` as a param to scan previos node states.

I would lean to longer params/names that are obvious, rather than shorter once that look nice.

Reply:
I see what you're saying. So `show_deleted` would basically ignore the `deactivated_at` value and `include_all_revisions` would iterate over all of the node revisions to pull the materialization. Let me update this.

Reply:
Updated 👍

Reply:
Yes, i think this now supports all the query options. Thanks!